### PR TITLE
Instance: Allow nosymfollow mount flag for container apparmor profile

### PR DESCRIPTION
**Pull request by mihalicyn**
Closes https://github.com/canonical/lxd/pull/12698

### CODE REVIEW

**Comment by mihalicyn** [comment]
cc @enr0n

**Comment by tomponline** [comment]
@mihalicyn please can you add what this fix resolves in terms of which ubuntu version we can now run as containers?

**Comment by mihalicyn** [comment]
> @mihalicyn please can you add what this fix resolves in terms of which ubuntu version we can now run as containers?

I'm not sure, because if I understand correcly this issue is only reproducible with some experimental systemd versions and with Oracular. I wasn't able to reproduce any issues locally. I hope @enr0n will help us with testing/validation of this.

File: lxd/apparmor/feature_check.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  #include <abstractions/base>
+  #include <abstractions/consoles>
+  #include <abstractions/nameservice>
+
+{{- if eq .feature "mount_nosymfollow" }}
+  mount options=(nosymfollow) /,
+{{- end }}
+
+}
+`))
```
Comment (simondeziel) [inline lxd/apparmor/feature_check.go:27]:
```suggestion
var featureCheckProfileTpl = template.Must(template.New("featureCheckProfile").Parse(`#
profile "{{ .name }}" flags=(attach_disconnected,mediate_deleted) {

{{- if eq .feature "mount_nosymfollow" }}
  mount options=(nosymfollow) /,
{{- end }}

}
`))
```

It's a nitpick but I'd prefer going with an even simpler profile as those abstractions are quite large (they include other files):

```
$ apparmor_parser --preprocess aa.profile | wc -l
1021
$ apparmor_parser --preprocess aa-light.profile | wc -l
3
```

**Review by simondeziel — APPROVED** [review]

File: lxd/apparmor/feature_check.go (hunk 1 above)
Comment (mihalicyn) [reply lxd/apparmor/feature_check.go]: Thanks, fixed!

**Review by simondeziel — APPROVED** [review]

File: lxd/sys/os.go [hunk 2]
```diff
@@ -69,6 +75,7 @@ type OS struct {
 	AppArmorConfined  bool
 	AppArmorStacked   bool
 	AppArmorStacking  bool
+	AppArmorFeatures  AppArmorFeaturesInfo
```
[the remark this answers is not in the document]
Comment (tomponline) [reply lxd/sys/os.go:78]: We'd need to move the other AppArmor* fields into this new sub-struct.

**Comment by tomponline** [comment]
@mihalicyn @enr0n has advised:

> The quickest thing to check is systemctl status systemd-resolved (or systemd-networkd, systemd-journald, or many others).
> Without the fix, they will fail with 243/CREDENTIALS
> On oracular with systemd from oracular-proposed
> So systemd v256 essentially

**Comment by mihalicyn** [comment]
Tested and it does not work. Looks like we need something else in addition to this change.

**Comment by tomponline** [comment]
> Tested and it does not work. Looks like we need something else in addition to this change.

Lets move this back to draft.

**Review by tomponline — APPROVED** [review]

**Pull request merged**